backend-go/config.go

```diff
@@ -1,232 +1,104 @@
-func updateWith[V any | string](base *map[string]any, updates *map[string]V) {
-	for k, v := range *updates {
-		if _, contained := (*base)[k]; !contained {
-			(*base)[k] = v
+// CliArgsMap returns a map of all CLI arguments in the form of --KEY=VALUE or --KEY VALUE
+// where KEY is the uppercase key and VALUE is the value.
+// If the argument is a flag (i.e., does not have a value), the value is true.
+// Positional arguments are ignored.
+func CliArgsMap() map[string]any {
+	res := make(map[string]any)
+	for /* don't include executable ($0) => start at 1 */ i := 1; i < len(os.Args); i++ {
+		key := os.Args[i]
+
+		if !strings.HasPrefix(key, "--") {
+			// ignore positional arguments
+			continue
 		}
-	}
-}
-
-// Parses the console arguments and returns a map that holds the configuration parameters.
-func cliConfig() *map[string]any {
-	// setup flags
-	var (
-		dev          bool
-		natsUrl      string
-		natsUser     string
-		natsPassword string
-		configFile   string
-		configKey    string
-		serviceName  string
-		dataDir      string
-	)
-
-	flag.BoolVar(&dev, "dev", false, "If set, program will start in development mode")
-
-	flag.StringVar(&natsUrl, "NATS_URL", "", "NATS url of the server the service can connect to")
-	flag.StringVar(&natsUser, "NATS_USER", "", "NATS user name for the authentication with the server")
-	flag.StringVar(&natsPassword, "NATS_PASSWORD", "", "NATS password for the authentication with the server "+
-		"(Note: It is recommended to set this via the environment variables or the config!)")
-
-	flag.StringVar(&configFile, "CONFIG_FILE", "", "file path to the config of the service")
-	flag.StringVar(&configKey, "CONFIG_KEY", "", "object key of a config file")
-
-	flag.StringVar(&serviceName, "SERVICE_NAME", "", "name of the service also used in the nats service "+
-		"registration")
-	flag.StringVar(&dataDir, "DATA_DIR", "", "path where the service can store persistent data")
-
-	// we don't really like the default message of the flag package
-	flag.Usage = func() {
-		fmt.Printf("Usage: %s [options] [field_0 ... field_n]\n\nParameters:\n", os.Args[0])
-		flag.PrintDefaults()
-	}
-	flag.Parse()
-
-	flagValues := map[string]any{
-		"NATS_URL":      natsUrl,
-		"NATS_USER":     natsUser,
-		"NATS_PASSWORD": natsPassword,
-		"CONFIG_FILE":   configFile,
-		"CONFIG_KEY":    configKey,
-		"SERVICE_NAME":  serviceName,
-		"DATA_DIR":      dataDir,
-	}
-
-	// prepare output map
-	parsedArgs := map[string]any{
-		"DEV": dev,
-	}
 
-	// only populate parsedArgs with entries that were, indeed, given (dev is an exception)
-	flag.Visit(func(currentFlag *flag.Flag) {
-		if value, ok := flagValues[currentFlag.Name]; ok {
-			parsedArgs[currentFlag.Name] = value
+		key = strings.TrimPrefix(key, "--")
+		if strings.Contains(key, "=") {
+			// --KEY=VALUE
+			split := strings.Split(key, "=")
+			innerKey := strings.ToUpper(split[0]) // all config keys are uppercase
+			innerValue := strings.Join(split[1:], "=")
```

Comment:
How? It can't be a part of the key (as they have to be environment variable compatible), and for the content, thanks to joining them back together with (with `"="`), that shouldn't be an issue, I think?

Reply:
Oops, I missed the `strings.Join()`. Sorry.

Reply:
No worries :)